### PR TITLE
Feature/syslog structured data

### PR DESCRIPTION
**Pull request by sametd**
This PR implements the "origin" component of the syslog structural data as described in RFC5424, specifically in sections 6.2 and 7.2 ([RFC5424 Section 7.2](https://datatracker.ietf.org/doc/html/rfc5424#section-7.2)). Implementation also ensures that the current functionality remains unaffected if the newly introduced variables are not set. I am tagging @tlmquintino here as he may also want to take a look at this implementation.

Note: This is not a full implementation of structural data of the syslog, only implements the bits that necessary to add logserver-compatible logging. Logserver repo can be found [here](https://github.com/ecmwf/logserver-ansible)

### CODE REVIEW

File: src/eckit/log/SysLog.h [hunk 1]
```diff
@@ -99,6 +99,11 @@ class SysLog {
         return s;
     }
 
+    /// Optional fields for structured data (RFC 5424 section 6.3)
+    void software(const std::string& software) { software_ = software; }
```
Comment (simondsmart) [inline src/eckit/log/SysLog.h:103]: Where are these being set?

**Comment by simondsmart** [comment]
Looks uncontroversial. Why is there a failing test?

File: src/eckit/log/SysLog.h (hunk 1 above)
Comment (sametd) [reply src/eckit/log/SysLog.h:103]: These are the keywords that belongs to origin part of the syslog structural data described in RFC5424 -> https://datatracker.ietf.org/doc/html/rfc5424#section-7.2

**Review by simondsmart — APPROVED** [review]
Looks good to me. Once the CI is happy, we should be good to go.

**Pull request merged**